diff --git a/mercurial/hgweb/server.py b/mercurial/hgweb/server.py
index a14bc75..c8d02e6 100644
--- a/mercurial/hgweb/server.py
+++ b/mercurial/hgweb/server.py
@@ -182,7 +182,7 @@ class _shgwebhandler(_hgwebhandler):
     def do_write(self):
         from OpenSSL.SSL import SysCallError
         try:
-            super(_shgwebhandler, self).do_write()
+            _hgwebhandler.do_write(self)
         except SysCallError, inst:
             if inst.args[0] != errno.EPIPE:
                 raise
@@ -190,7 +190,7 @@ class _shgwebhandler(_hgwebhandler):
     def handle_one_request(self):
         from OpenSSL.SSL import SysCallError, ZeroReturnError
         try:
-            super(_shgwebhandler, self).handle_one_request()
+            _hgwebhandler.handle_one_request(self)
         except (SysCallError, ZeroReturnError):
             self.close_connection = True
             pass
@@ -278,7 +278,7 @@ def create_server(ui, repo):
         def __init__(self, *args, **kwargs):
             if self.address_family is None:
                 raise error.RepoError(_('IPv6 is not available on this system'))
-            super(IPv6HTTPServer, self).__init__(*args, **kwargs)
+            MercurialHTTPServer.__init__(self, *args, **kwargs)
 
     if ssl_cert:
         handler = _shgwebhandler
