diff --git a/mercurial/cmdutil.py b/mercurial/cmdutil.py
index 41ffdef..a5461c6 100644
--- a/mercurial/cmdutil.py
+++ b/mercurial/cmdutil.py
@@ -389,7 +389,7 @@ def copy(ui, repo, pats, opts, rename=False):
             return
 
         # check for overwrites
-        exists = os.path.exists(target)
+        exists = os.path.lexists(target)
         if not after and exists or after and state in 'mn':
             if not opts['force']:
                 ui.warn(_('%s: not overwriting - file exists\n') %
